### PR TITLE
Sort application schemes before other auto generated schemes

### PR DESCRIPTION
**Pull request by David-Lam**
This pr improve developer experience by making Xcode select the app scheme by default when generating xcodeproj for multiple top level targets.

### CODE REVIEW

File: tools/generators/legacy/src/Generator/CreateAutogeneratedXCSchemes.swift [hunk 1]
```diff
@@ -120,7 +121,8 @@ extension Generator {
                         buildMode: buildMode,
                         schemeInfo: schemeInfo
                     ),
-                    isTopLevel: pbxTarget.isTopLevel
+                    isTopLevel: pbxTarget.isTopLevel,
+                    isApplication: pbxTarget.productType == PBXProductType.application
```
Comment (luispadron) [inline tools/generators/legacy/src/Generator/CreateAutogeneratedXCSchemes.swift:125]: nit: Make extension like `pbxTarget.isTopLevel`

**Review by luispadron — APPROVED** [review]

**Review by yongjincho92 — APPROVED** [review]

**Comment by David-Lam** [comment]
Good call! I think in this case we don't need to check `isTopLevel` since the product type order should already imply top level product type before non top levels.

File: tools/generators/legacy/src/Generator/CreateAutogeneratedXCSchemes.swift (hunk 1 above)
Comment (David-Lam) [reply tools/generators/legacy/src/Generator/CreateAutogeneratedXCSchemes.swift]: Thanks for the suggestion! [Moved logic to extension file.](https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2211/files#diff-176ddd00ccbdfa85509edd61a4e12ad0ca0d6d51e6b16acb57f2753a3cea2618R284-R328)

**Review by luispadron — APPROVED** [review]

**Comment by brentleyjones** [comment]
I wasn't clear, but I don't think a strict ordering like you did is best. I think it should be groups like I suggested, where types in the same group would be ordered by name. So a command line tool could be listed before an app depending on its name.

**Comment by David-Lam** [comment]
Sounds good! I misunderstand the previous suggestion. [Updated](https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2211/files#diff-176ddd00ccbdfa85509edd61a4e12ad0ca0d6d51e6b16acb57f2753a3cea2618R286-R317) to set product type ordering as group.

File: tools/generators/legacy/src/Extensions/PBXProductType+Extensions.swift [hunk 2]
```diff
@@ -281,4 +281,40 @@ extension PBXProductType {
...
+        switch self {
+        // Applications
+        case .application,
+            .onDemandInstallCapableApplication,
+            .watchApp,
+            .watch2App,
+            .watch2AppContainer,
+            .commandLineTool,
+            .messagesApplication,
+            .xpcService:
```
Comment (brentleyjones) [inline tools/generators/legacy/src/Extensions/PBXProductType+Extensions.swift:295]: nit:
```suggestion
        case .application,
            .commandLineTool,
            .messagesApplication,
            .onDemandInstallCapableApplication,
            .watchApp,
            .watch2App,
            .watch2AppContainer,
            .xpcService:
```

File: tools/generators/legacy/src/Extensions/PBXProductType+Extensions.swift [hunk 3]
```diff
@@ -281,4 +281,40 @@ extension PBXProductType {
...
+            .driverExtension,
+            .systemExtension,
+            .messagesExtension,
+            .stickerPack,
+            .xcodeExtension:
+            return 1
+        // Tests
+        case .unitTestBundle,
+            PBXProductType.uiTestBundle,
+            PBXProductType.ocUnitTestBundle:
```
Comment (brentleyjones) [inline tools/generators/legacy/src/Extensions/PBXProductType+Extensions.swift:313]: nit:
```suggestion
        case .ocUnitTestBundle,
            .uiTestBundle,
            .unitTestBundle:
```

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**